src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -62,6 +64,22 @@ export class DatasetService {
     return this.defaultDataset;
   }
 
+  public getRecentDataset(dataset: string | undefined): Dataset | undefined {
```

Comment:
```suggestion
  public getRecentDataset(id: string | undefined): Dataset | undefined {
```
or `datasetId` passing in a  dataset and getting a dataset doesnt make sense right?